### PR TITLE
Implement datastore and saving option

### PR DESCRIPTION
**Pull request by ninaprabhu**

### CODE REVIEW

**Comment by Tetheta** [comment]
Starting review

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -29,22 +32,22 @@
...
     for(String s:array) {
-        comments.add(s);
+        Entity comment = new Entity("Comment");
+        comment.setProperty("text", s);
+        comment.setProperty("timestamp", System.currentTimeMillis());
+        datastore.put(comment);
     }
-    response.setContentType("application/json;");
-    response.getWriter().println(comments);
+    long timestamp = System.currentTimeMillis();
```
Comment (Tetheta) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:51]: Shouldn't this either not be here or be declared above the for loop and used? The latter would probably be ideal so you don't call into the API every time.

Comment (ninaprabhu) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Good point, just fixed to reflect that

**Comment by ninaprabhu** [comment]
Sorry for pushing more changes to this PR - I probably should have made another branch. Apologies for spacing on that!

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -26,7 +26,9 @@ <h1>About me!</h1>
     <form id="java" action="/data" method="POST">
       <p>Enter your comments (separate by commas):</p>
       <textarea name="text-container">Sample</textarea>
-      <br/><br/>
+      <br/>
+      <input type="checkbox" name="save" value="true">Save comments
```
Comment (Tetheta) [inline portfolio/src/main/webapp/index.html:30]: Need a closing </input> tag for this I believe.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -29,21 +35,34 @@
...
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("text"); //we only want the comment text
+      comments.add(comment);
+    }
+
     response.setContentType("application/json;");
     response.getWriter().println(comments);
```
Comment (Tetheta) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:67]: Interesting that they have you responding with comment text in a POST request. Typically in prod you'd only respond with ids to fetch the data, but this works for now.

File: portfolio/src/main/webapp/index.html (hunk 2 above)
Comment (ninaprabhu) [reply portfolio/src/main/webapp/index.html:30]: Looking at resources I think input doesn't need a closing tag, but there's a label tag that can be used to hold the text like "save comments" that requires opening/closing. I'm looking into that right now!

**Pull request merged**